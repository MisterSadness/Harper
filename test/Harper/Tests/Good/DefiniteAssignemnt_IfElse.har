f :: Integer -> Integer;
f n = {
  var (x :: Integer);
  var (result :: Integer) = 0;
  
  if (n == 42) {
    x := n;
  
    // x is definitely assigned here.
  
    result += x;
  }
  else {
    x := 0;

    // x is definitely assigned here.

    result += x;
  }

  // x is definitely assigned here.

  result += x + n;
  return result;
};

main :: Integer;
main = f 42;