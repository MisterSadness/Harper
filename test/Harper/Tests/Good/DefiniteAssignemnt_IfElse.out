Execution ended with value: 126